### PR TITLE
Add E2E tests in which deployed instances use the region's shared net

### PR DESCRIPTION
**Pull request by pratid**
#### Reviewers

@flopezag @jframos 
#### Description

This provides the feature described in task 5095 (check that every region has a Shared Network in which we can deploy the VMs without creation of private networks and routers):
- The name of the shared network may be provided in settings file (a default value "shared-net" is used if not)
- E2E tests (ssh & SNAT) have been replicated into two new tests using the shared network instead of private networks.
#### Testing

Verified in CI environment with region Spain2 and "node-int-net-01" shared network

### CODE REVIEW

File: fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py [hunk 1]
```diff
@@ -160,6 +157,142 @@ def __get_external_network_test_helper__(self):
...
+    def __get_shared_network_test_helper__(self):
+        """
+        HELPER. Finds and returns the shared network name of current region
+        :return: Shared network name
+        """
+        shared_network_name = TEST_SHARED_NET_DEFAULT
+        shared_network_conf = self.conf[PROPERTIES_CONFIG_REGION].get(PROPERTIES_CONFIG_REGION_SHARED_NET)
+        if shared_network_conf:
+            shared_network_name = shared_network_conf.get(self.region_name, TEST_SHARED_NET_DEFAULT)
+        self.assertIsNotNone(shared_network_name, "No shared network found")
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py]: This assertion should never be executed but, if it is, add as part of the message that the shared network has not been found _in the configuration file_.

Comment (pratid) [inline fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py]: True. In fact, this assertion is an advance for future tests in which Neutron will be queried fro shared networks. At this moment it is useless, so we will remove it.

File: fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py [hunk 2]
```diff
@@ -347,116 +480,37 @@ def test_deploy_instance_with_network_and_associate_public_ip(self):
         # Associate Public IP to Server
         self.nova_operations.add_floating_ip_to_instance(server_id=server_id, ip_address=allocated_ip)
 
-    def test_deploy_instance_with_networks_and_e2e_connection_using_public_ip(self):
+    def test_deploy_instance_with_new_network_and_e2e_connection_using_public_ip(self):
         """
-        Test whether it is possible to deploy and instance, assign an allocated public IP and establish a SSH connection
+        Test whether it is possible to deploy an instance with new network, assign an allocated public IP
+        and establish a SSH connection
```
Comment (jframos) [inline fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py:485]: You are writing the description of the test using two lines but the `nosetests` lib does not process the second one (only the first one). Then, the _tooltip_ on the Dashboard's Test Detail page will not show the whole description...

**Comment by jframos** [comment]
Please, update the user documentation with the list of implemented/modified [test cases](doc/test_cases.rst)

**Comment by jframos** [comment]
LGTM

**Comment by flopezag** [comment]
LGTM

**Pull request merged**